docs/messaging/rabbitmq-client-integration.md

```diff
@@ -1,104 +1,214 @@
+
+builder.AddProject<Projects.ExampleProject>()
+        .WithReference(rabbitmq);
+
+// After adding all resources, run the app...
+```
+
+The data bind mount is used to persist the RabbitMQ server data across container restarts. The data bind mount is mounted at the `/mnt/rabbitmq` path in the RabbitMQ server container. For more information on data bind mounts, see [Docker docs: Bind mounts](https://docs.docker.com/engine/storage/bind-mounts).
+
+<!--
+TODO: Were are still missing the following bits:
+
+- Container lifetimes, link to related docs.
+- Health checks... in hosting, different than client.
+-->
+
+## Client integration
+
+<!--
+We need to also call out any related extension methods per/type of integration.
+-->
+
+To get started with the .NET Aspire RabbitMQ client integration, install the [Aspire.RabbitMQ.Client](https://www.nuget.org/packages/Aspire.RabbitMQ.Client) NuGet package in the client-consuming project, i.e., the project for the application that uses the RabbitMQ client. The RabbitMQ client integration registers an [IConnection](https://rabbitmq.github.io/rabbitmq-dotnet-client/api/RabbitMQ.Client.IConnection.html) instance that you can use to interact with RabbitMQ.
 
 ### [.NET CLI](#tab/dotnet-cli)
 
 ```dotnetcli
-dotnet add package Aspire.Hosting.RabbitMQ
+dotnet add package Aspire.RabbitMQ.Client
 ```
 
 ### [PackageReference](#tab/package-reference)
 
 ```xml
-<PackageReference Include="Aspire.Hosting.RabbitMQ"
+<PackageReference Include="Aspire.RabbitMQ.Client"
                   Version="[SelectVersion]" />
 ```
 
 ---
 
-In your app host project, register a RabbitMQ server and consume the connection using the following methods, such as <xref:Aspire.Hosting.RabbitMQBuilderExtensions.AddRabbitMQ%2A>:
+For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).
+
+### Add RabbitMQ client
+
+In the _:::no-loc text="Program.cs":::_ file of your client-consuming project, call the <xref:Microsoft.Extensions.Hosting.AspireRabbitMQExtensions.AddRabbitMQClient%2A> extension method to register an `IConnection` for use via the dependency injection container. The method takes a connection name parameter.
 
 ```csharp
-var builder = DistributedApplication.CreateBuilder(args);
+builder.AddRabbitMQClient(connectionName: "messaging");
+```
 
-var messaging = builder.AddRabbitMQ("messaging");
+You can then retrieve the `IConnection` instance using dependency injection. For example, to retrieve the connection from an example service:
 
-builder.AddProject<Projects.ExampleProject>()
-       .WithReference(messaging);
+```csharp
+public class ExampleService(IConnection connection)
+{
+    // Use connection...
+}
 ```
 
-The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method configures a connection in the `ExampleProject` project named `messaging`.
+### Add keyed RabbitMQ client
```

Comment:
I wonder if the "keyed" section should go lower. Using multiple isn't necessarily the 2nd thing you'd do when you want to use RabbitMQ.

Reply:
While I tend to agree it wouldn't we the 2nd thing you'd do, it might be the first though, so if a single DI registration call isn't sufficient, maybe you'd like to know if you could do keyed DI. They seem very related, you're doing one or the other is my point - so if they're mutually exclusive should we have them in tabs? Register Single Client or Register Multiple Clients? Or is that overkill?